### PR TITLE
Add CIFAR-10 Classification Results

### LINKED ISSUE #8: CIFAR-10 Sequence Classification
Implement and check CIFAR-10 Results vs. original implementation.

### PR DESCRIPTION
**Pull request by siddk**
Add pipeline for CIFAR-10 Classification, as well as smoothing out training loop kinks. Additionally updates some of the exposition in the literate `s4.py` implementation.

Making a Draft PR because current code **fails to replicate** results in [original S4 codebase](https://github.com/HazyResearch/state-spaces/blob/feeab742e9c737c8e2b8b0e44d3efff4049f5847/example.py#L20). Running apples-to-apples (or as close as possible) results in ~66% accuracy (vs. 89% reported).

Possible reasons:
- Lack of custom LR per S4 Parameters (see #16 -- [this function](https://github.com/HazyResearch/state-spaces/blob/feeab742e9c737c8e2b8b0e44d3efff4049f5847/example.py#L235)).
- Lack of LR Schedule (LR decay on validation/test accuracy plateau).

Resolves #8

### CODE REVIEW

**Comment by siddk** [comment]
Merge CIFAR --> just folded in the Optimization commits.

**Comment by siddk** [comment]
Updated results, run command is [here](https://github.com/srush/annotated-s4/pull/17/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R99).

Resolves #8

**Comment by siddk** [comment]
@srush - right now the replication command is just in the README; do we want to get rid of the unnecessary (other hyperparameter) runs?

**Comment by srush** [comment]
Yes, think we should clean up the readme to the working commands.

**Comment by siddk** [comment]
Done
